src/convnets/shufflenet.jl

```diff
@@ -0,0 +1,160 @@
+    end
+    return x
+end
+
+"""
+ShuffleUnit(in_channels::Integer, out_channels::Integer, grps::Integer, downsample::Bool, ignore_group::Bool)
+
+Shuffle Unit from 'ShuffleNet: An Extremely Efficient Convolutional Neural Network for Mobile Devices
+([reference](https://arxiv.org/abs/1707.01083)).
+
+# Arguments
+
+  - `in_channels`: number of input channels
+  - `out_channels`: number of output channels
+  - `groups`: number of groups
+  - `downsample`: apply downsaple if true
+  - `ignore_group`: ignore group convolution if true
+"""
+function ShuffleUnit(in_channels::Integer, out_channels::Integer, groups::Integer, downsample::Bool, ignore_group::Bool)
+    mid_channels = out_channels ÷ 4
+    groups = ignore_group ? 1 : groups
+    strd = downsample ? 2 : 1
+
+    if downsample
+        out_channels -= in_channels
+    end
+
+    m = Chain(Conv((1,1), in_channels => mid_channels; groups,pad=SamePad()),
+              BatchNorm(mid_channels),
+              NNlib.relu,
+              x -> ChannelShuffle(x, groups),
+              DepthwiseConv((3,3),  mid_channels => mid_channels; bias=false, stride=strd, pad=SamePad()),
+              BatchNorm(mid_channels),
+              NNlib.relu,
+              Conv((1,1), mid_channels => out_channels; groups, pad=SamePad()),
+              BatchNorm(out_channels),
+              NNlib.relu)
+    
+    if downsample
+        m = Parallel((mx, x) -> cat(mx, x, dims=3),m, MeanPool((3,3); pad=SamePad(), stride=2))
+    else
+        m = SkipConnection(m, +)
+    end
+    return m
+end
+
+"""
+ShuffleNet(channels, init_block_channels::Integer, groups, num_classes; in_channels=3)
+
+ShuffleNet model from 'ShuffleNet: An Extremely Efficient Convolutional Neural Network for Mobile Devices
+([reference](https://arxiv.org/abs/1707.01083)).
+
+# Arguments
+
+  - `channels`: list of channels per layer
+  - `init_block_channels`: number of output channels from the first layer
+  - `groups`: number of groups
+  - `num_classes`: number of classes
+  - `in_channels`: number of input channels
+"""
+function ShuffleNet(channels, init_block_channels::Integer, groups, num_classes; in_channels=3)
+    features = []
+
+    append!(features, [Conv((3,3), in_channels => init_block_channels; stride=2, pad=SamePad()),
+    BatchNorm(init_block_channels),
+    NNlib.relu,
+    MaxPool((3,3); stride=2, pad=SamePad())])
+
+    in_channels::Integer = init_block_channels
+  
+    for (i, num_channels) in enumerate(channels)
+        stage = []
+        for (j, out_channels) in enumerate(num_channels)
+            downsample = j==1
+            ignore_group = i==1 && j==1
+            out_ch::Integer = trunc(out_channels)
+            push!(stage, ShuffleUnit(in_channels, out_ch, groups, downsample, ignore_group))
+            in_channels = out_ch
+        end
+        append!(features, stage)
+    end
+  
+    model = Chain(features...)
+  
+    return Chain(model, GlobalMeanPool(), Flux.flatten, Dense(in_channels => num_classes))
```

Comment:
```suggestion
    return Chain(model, GlobalMeanPool(), MLUtils.flatten, Dense(in_channels => num_classes))
```
The general modus operandi of this library has been to create named types for the top-level model and wrap the underlying Chain with them. You can see this pattern in the files for any of the other exported models.

For the suggestion. `flatten` is only imported and not defined in Flux. It's preferable to use a symbol from the library that actually defined when that library is available (which MLUtils is, being a dep of Metalhead).

Reply:
Could I see an example? Sorry, I'm still a newbie using Julia, I looked to the rest of convnets and tried to code with a similar style, but there are still things I that still haven't fully understood.